File: erpnext/selling/report/customer_report/customer_report.py
# Copyright (c) 2013, Frappe Technologies Pvt. Ltd. and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe


def execute(filters=None):
        columns = get_columns(filters)
        data = get_data(filters)
        return columns, data

def get_data(filters):
        cond = " 1= 1"

        if filters.get("dzongkhag"):
                cond += " and dzongkhag = '{0}'".format(filters.get("dzongkhag"))

        if filters.get("from_date") and filters.get("to_date"):
                cond += " and creation between '{0}' and '{1}'".format(filters.from_date, filters.to_date)

	#and customer_group != 'Internal'
        query = frappe.db.sql(""" select name, customer_name, customer_id, customer_type, customer_group, territory, 
                        location, dzongkhag,  date(creation), mobile_no from `tabCustomer`  where {0}  order by name asc""".format(cond), debug =1)
        return query



def get_columns(filters):
        cols = [
                ("Customer ID") + ":Link/Customer:120",
                ("Customer Name") + ":Data:180",
                ("Customer CID") + ":Data:120",
                ("Customer Type") + ":Data:130",
                ("Customer Group") + ":Data:130",
                ("Teritory") + ":Data:100",
                ("Location") + ":Data:120",
                ("Dzongkha") + ":Data:120",
                ("Date of Creation") + ":Date:120",
                ("Contact Number") + ":Data:120"
        ]
        return cols

File: erpnext/selling/report/customer_report/customer_report.js
// Copyright (c) 2016, Frappe Technologies Pvt. Ltd. and contributors
// For license information, please see license.txt
frappe.query_reports["Customer Report"] = {
	"filters": [
		{
                        "fieldname": "dzongkhag",
                        "label": ("Dzongkhag"),
                        "fieldtype": "Link",
                        "width": "80",
                        "options": "Dzongkhags",
                },
                {
                        "fieldname": "customer",
                        "label": ("Customer"),
                        "fieldtype": "Link",
                        "width": "80",
                        "options": "Customer",
                },
                {
                        "fieldname": "customer_group",
                        "label": ("Customer Group"),
                        "fieldtype": "Link",
                        "width": "80",
                        "options": "Customer Group",
                },
                {
                        "fieldname": "construction_type",
                        "label": ("Construction Type"),
                        "fieldtype": "Link",
                        "width": "80",
                        "options": "Construction Type",
                },
                {
                        "fieldname": "branch",
                        "label": ("Branch"),
                        "fieldtype": "Link",
                        "width": "80",
                        "options": "Branch",
                },
                {
                        "fieldname": "territory",
                        "label": ("Region"),
                        "fieldtype": "Link",
                        "width": "80",
                        "options": "Territory",
                },
                {
                        "fieldname": "item_group",
                        "label": ("Material Group"),
                        "fieldtype": "Link",
                        "options": "Item Group",
                        "get_query": function() {
                                return {"doctype": "Item Group", "filters": {"is_group": 0, "is_production_group": 1}}
                        },
                },
                {
                        "fieldname": "item_sub_group",
                        "label": ("Material Sub Group"),
                        "fieldtype": "Link",
                        "options": "Item Sub Group",
                        "get_query": function() {
                                var item_group = frappe.query_report.filters_by_name.item_group.get_value();
                                return {"doctype": "Item Sub Group", "filters": {"item_group": item_group}}
                        }
                },
                {
                        "fieldname": "item",
                        "label": ("Material"),
                        "fieldtype": "Link",
                        "options": "Item",
                        "get_query": function() {
                                var sub_group = frappe.query_report.filters_by_name.item_sub_group.get_value();
                                return {"doctype": "Item", "filters": {"item_sub_group": sub_group, "is_production_item": 1}}
                        }
                },
                {
			"fieldname": "warehouse",
			"label": __("Warehouse"),
			"fieldtype": "Link",
			"width": "80",
                        "options":"Warehouse"
                },
                {
                        "fieldname":"from_date",
                        "label": ("From Date"),
                        "fieldtype": "Date",
                        "width": "80",
                        "default": sys_defaults.year_start_date,
                },
                {
                        "fieldname":"to_date",
                        "label": ("To Date"),
                        "fieldtype": "Date",
                        "width": "80",
                        "default": frappe.datetime.get_today()
                },
                {
                        "fieldname":"aggregate",
                        "label": ("Show Aggregate"),
                        "fieldtype": "Check",
                        "default": 0
                }
	]
}
